robot-name/ena/robot_name.go

```diff
@@ -0,0 +1,57 @@
+package robotname
+
+import (
+	"errors"
+	"math"
+	"math/rand"
+	"time"
+)
+
+var usedNames = make(map[string]bool)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+type Robot struct {
+	name string
+}
+
+func (r *Robot) Name() (string, error) {
+	const letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	const lettersPartLen = 2
+	const digits = "0123456789"
+	const digitsPartLen = 3
+
+	if r.name != "" {
+		return r.name, nil
+	}
+
+	allPossibleNames := int(math.Pow(float64(len(letters)), lettersPartLen) * math.Pow(float64(len(digits)), digitsPartLen))
```

Comment:
This should be a constant.

Reply:
I tried to make it const initially, but it fails with the following error `./robot_name.go:30:8: const initializer int(math.Pow(float64(len(letters)), lettersPartLen) * math.Pow(float64(len(digits)), digitsPartLen)) is not a constant`. Did not spend time to understand why.

Reply:
You cannot use math.Pow in constants (mostly function calls):

> A constant value is represented by a rune, integer, floating-point, imaginary, or string literal, an identifier denoting a constant, a constant expression, a conversion with a result that is a constant, or the result value of some built-in functions such as unsafe.Sizeof applied to any value, cap or len applied to some expressions, real and imag applied to a complex constant and complex applied to numeric constants. 

You can though hardcode the result of the operation.